src/Sarif/Cache.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    ///  Cache is a simple LRU (least recently used) collection.
+    ///  The base class libraries don't offer one without an extra dependency.
+    /// </summary>
+    /// <typeparam name="TKey">Type of Keys in collection</typeparam>
+    /// <typeparam name="TValue">Type of Values being cached in collection</typeparam>
+    public class Cache<TKey, TValue> where TKey : IComparable<TKey>
+    {
+        private readonly Func<TKey, TValue> _builder;
+        private readonly Dictionary<TKey, TValue> _cache;
+        private readonly LinkedList<TKey> _keysInUseOrder;
+
+        public const int DefaultCapacity = 100;
+        public int Capacity { get; private set; }
```

Comment:
>private set; [](start = 35, length = 12)

Since you set this only in the ctor, you don't need `private set`. (C# 6 feature: you can set a readonly property in the ctor.)

Reply:
Huh. Ok, cool. I tried marking it "readonly" and it told me the language version wasn't late enough. I didn't think about trying to remove the setter, but that works.

Reply:
No, sorry, a `reaonly` property is one with only a getter. You can remove the private setter because it's ok to set such a property from the ctor even tho it has no setter.

---
In reply to: [382333251](https://github.com/microsoft/sarif-sdk/pull/1786#discussion_r382333251) [](ancestors = 382333251)